src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ModuleInitializerAttributeShouldNotBeUsedInLibraries.cs

```diff
@@ -0,0 +1,80 @@
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.ModuleInitializerAttributeShouldNotBeUsedInLibrariesMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.ModuleInitializerAttributeShouldNotBeUsedInLibrariesDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                    s_localizableTitle,
+                                                                    s_localizableMessage,
+                                                                    DiagnosticCategory.Usage,
+                                                                    RuleLevel.BuildWarning,
+                                                                    s_localizableDescription,
+                                                                    isPortedFxCopRule: true,
+                                                                    isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                INamedTypeSymbol? moduleInitializerAttributeType = compilationContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeCompilerServicesModuleInitializerAttribute);
+
+                if (moduleInitializerAttributeType is null) return;
+
+                compilationContext.RegisterSymbolAction(context =>
+                {
+                    if (context.Symbol is IMethodSymbol method)
+                    {
+                        if (method.IsPrivate() ||
+                            method.Parameters.Length > 0 ||
+                            method.IsGenericMethod ||
+                            !method.IsStatic ||
+                            !method.ReturnsVoid
+                        )
+                        {
+                            return;
+                        }
+
+                        AttributeData? initializerAttribute = context.Symbol.GetAttributes(moduleInitializerAttributeType).FirstOrDefault();
```

Comment:
`AttributeData.ApplicationSyntaxReference` is annotated as nullable, and I see other analyzers check for null. Can this happen for this analyzer?

Reply:
Thanks; I'll guard it.